### DISCUSSION TITLE
Is imgproxy secure against CVE-2023-4863?

### DISCUSSION THREAD

**Discussion by sihu** [imgproxy/imgproxy#1237] [Q&A] [answered]
Since I could not find any mention of CVE-2023-4863 (described in this [blog](https://blog.cloudflare.com/uncovering-the-hidden-webp-vulnerability-cve-2023-4863)) in this repo I was wondering, if it was secure against it? 

I saw that webp images are limited in these lines https://github.com/imgproxy/imgproxy/blob/a2fd4bc063f65595a73082e5de8be360c85c2087/processing/fix_size.go#L14-L36 but i am not sure if all webp images are processed since at least animated in the docs this was mentioned:

> imgproxy can process animated images (GIF, WebP), but since this operation is pretty memory heavy, only one frame is processed by default.

**Comment by DarthSim** [comment, marked as answer]
imgproxy uses libwebp 1.3.2 which has this CVE fixed

**Reply by sihu** [reply]
thx so much for the quick reply. just to ensure: 

if i serve a malicious webp through the proxy that was uploaded by someone, the clients will not be affected, even if they use an old chrome that has still an old version of libwebp in it?

**Reply by DarthSim** [reply]
Usage of libwebp 1.3.2 protects imgproxy itself. When imgproxy processes an image, it decodes it and then encodes it to the resulting format. So the resulting image won't be malicious anyway. if imgproxy doesn't process an image due to the `IMGPROXY_SKIP_PROCESSING_FORMATS` config or the `raw` option, it serves the original image without modifications.
